Objective-C/CBLCollection.mm

```diff
@@ -17,49 +17,122 @@
 //  limitations under the License.
 //
 
-#import "CBLCollection.h"
-#import "CBLIndexable.h"
 #import "CBLChangeListenerToken.h"
+#import "CBLCollection+Internal.h"
 #import "CBLCollectionChangeObservable.h"
+#import "CBLCoreBridge.h"
+#import "CBLDatabase+Internal.h"
+#import "CBLIndexable.h"
+#import "CBLIndexConfiguration+Internal.h"
+#import "CBLScope.h"
+#import "CBLScope+Internal.h"
+#import "CBLStatus.h"
+#import "CBLStringBytes.h"
+
+using namespace fleece;
 
 NSString* const kCBLDefaultCollectionName = @"_default";
 
-@implementation CBLCollection
+@implementation CBLCollection {
+    C4Collection* _c4col;
+}
 
-@synthesize count=_count, name=_name, scope=_scope;
+@synthesize count=_count, name=_name, scope=_scope, db=_db;
 
-- (instancetype) initWithName: (NSString*)name
-                        scope: (nullable CBLScope*)scope
-                        error: (NSError**)error {
-    CBLAssertNotNil(name);
-    
+- (instancetype) initWithDB: (CBLDatabase*)db
+               c4collection: (C4Collection*)c4collection
+                      error: (NSError**)error {
```

Comment:
The error is not needed here. I have the same comment for the CBLScope's init as it is only for passing to the CBLScope's init.